### PR TITLE
Windows support: Makefile changes

### LINKED ISSUE #2614: Windows support: Makefile changes
Current Makefile only supports `linux` and `darwin`. Provide support to Windows in all the targets of the Makefile.

### PR DESCRIPTION
**Pull request by MarcosDY**
Apply changes to Make file and scripts to be able to run them on windows,
all targets are working except for integration

**Which issue this PR fixes**
fixes #2614

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -97,8 +100,15 @@ build_dir := $(DIR)/.build/$(os1)-$(arch1)
 go_version_full := $(shell cat .go-version)
 go_version := $(go_version_full:.0=)
 go_dir := $(build_dir)/go/$(go_version)
-go_bin_dir := $(go_dir)/bin
-go_url = https://storage.googleapis.com/golang/go$(go_version).$(os1)-$(arch2).tar.gz
+
+ifeq ($(os1),windows)
+	go_bin_dir = $(go_dir)/go/bin
```
Comment (azdagron) [inline Makefile:105]: Why does this have to be different?

File: Makefile [hunk 2]
```diff
@@ -272,7 +284,11 @@ else
 endif
 
 integration:
+ifeq ($(os1), windows)
+	$(error Integration tests are no supported on windows)
```
Comment (azdagron) [inline Makefile:288]:
```suggestion
	$(error Integration tests are not supported on windows)
```

File: Makefile (hunk 1 above)
Comment (MarcosDY) [reply Makefile:105]: windows golang zip does not contains files on base, but they put all files inside `go` folder you can take a look to it downloading for example https://go.dev/dl/go1.17.4.windows-amd64.zip

Comment (azdagron) [reply Makefile:105]: 👍 ok, sounds good

**Review by azdagron — APPROVED** [review]

**Pull request merged**